clerk/genesis.go

```diff
@@ -23,12 +23,16 @@ func DefaultGenesisState() GenesisState {
 
 // InitGenesis sets distribution information for genesis.
 func InitGenesis(ctx sdk.Context, keeper Keeper, data GenesisState) {
+
 	// add checkpoint headers
-	if len(data.EventRecords) != 0 {
+	eventCount := len(data.EventRecords)
+	if eventCount != 0 {
 		for _, record := range data.EventRecords {
 			keeper.SetEventRecord(ctx, *record)
 		}
 	}
+
+	keeper.SetStateSyncEventCount(ctx, uint64(eventCount))
```

Comment:
It might be the case where `keeper.SetEventRecord(ctx, *record)` will return an error but eventCount will not reflect that. Let's handle that error and/or ensure `EventCount` is always equal to actual events stored.

Reply:
Event Count is always equal to actual events stored.

Reply:
oh yes, it has to be, but in this case in one case it might not be. The case where `SetStateSyncEventCount` throws an error.